app/javascript/components/concept-exercise-graph/hooks/useWebpageSize.ts

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from 'react'
+
+type WebpageSize = {
+  width: undefined | number
+  height: undefined | number
+}
+
+export function useWebpageSize() {
+  // Initialize state with undefined width/height so server and client renders match
+  // Learn more here: https://joshwcomeau.com/react/the-perils-of-rehydration/
+
+  const [webpageSize, setWebpageSize] = useState({
+    width: undefined,
+    height: undefined,
+  } as WebpageSize)
+
+  useEffect(() => {
+    function handleResize() {
+      setWebpageSize({
+        width: document.documentElement.clientWidth,
+        height: document.body.scrollHeight,
+      })
+    }
+
+    window.addEventListener('resize', handleResize)
```

Comment:
Is `window` just a globally accessible thing here?

Reply:
Ya, in the browser context it's a global

Reply:
Interesting. Good to know.